clients/drcachesim/tests/memref_gen.h

```diff
@@ -137,6 +145,41 @@ gen_exit(memref_tid_t tid)
     return memref;
 }
 
-} // namespace
+/* Returns a vector of encoded memref_t.
+ * The caller has to set tid + pid fields of the memref_t in memref_instr_t
+ * structs but not the other fields. Also note that all operand records have to
+ * be filled in for each memref when constructing memref_instr_vec.
+ */
+inline std::vector<memref_t>
+get_memrefs_from_ir(instrlist_t *ilist, std::vector<memref_instr_t> &memref_instr_vec,
+                    addr_t base_addr)
+{
+    static const int MAX_DECODE_SIZE = 1024;
+    byte decode_buf[MAX_DECODE_SIZE];
+    byte *pc =
+        instrlist_encode_to_copy(GLOBAL_DCONTEXT, ilist, decode_buf,
+                                 reinterpret_cast<app_pc>(base_addr), nullptr, true);
+    assert(sizeof(decode_buf) / sizeof(decode_buf[0]) <= MAX_DECODE_SIZE);
+    assert(pc != nullptr);
+    std::vector<memref_t> memrefs = {};
+    memrefs.push_back(
+        gen_marker(1, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS));
+    for (auto pair : memref_instr_vec) {
+        if (pair.instr != nullptr && type_is_instr(pair.memref.instr.type)) {
```

Comment:
As discussed offline, remove `pair.instr != nullptr &&` from this condition and assert that it is true in the if-body.

Reply:
That's why I suggested asserting inside the if-body where we know it's an instr record, not a data record.